addons/common/functions/fnc_claimSafeServer.sqf

```diff
@@ -35,19 +35,10 @@ if (!isNull _owner && {!isNull _unit} && {_unit != _owner}) exitWith {
 _target setVariable [QGVAR(owner), _unit, true];
 
 // lock target object
-if (_lockTarget) then {
-    private _canBeDisassembled = ([] isNotEqualTo getArray (configOf _target >> "assembleInfo" >> "dissasembleTo")) && { !([false, true] select (_target getVariable [QEGVAR(csw,assemblyMode), 0])) };
-    if (!isNull _unit) then {
-        [QGVAR(lockVehicle), _target, _target] call CBA_fnc_targetEvent;
-        if (_canBeDisassembled) then {
-            _target enableWeaponDisassembly false;
-        };
-    } else {
-        [QGVAR(unlockVehicle), _target, _target] call CBA_fnc_targetEvent;
-        if (_canBeDisassembled) then {
-            _target enableWeaponDisassembly true;
-        };
-    };
+if (_lockTarget && {!isNull _unit}) then {
+    [QGVAR(lockVehicle), _target, _target] call CBA_fnc_targetEvent;
+} else {
+    [QGVAR(unlockVehicle), _target, _target] call CBA_fnc_targetEvent;
```

Comment:
It isn't quite the same logic. Do you want a vehicle to be unlocked if `_unit` is `objNull`, despite `_lockTarget` being true?

```suggestion
if (_lockTarget) then {
    if (!isNull _unit) then {
        [QGVAR(lockVehicle), _target, _target] call CBA_fnc_targetEvent;
    } else {
        [QGVAR(unlockVehicle), _target, _target] call CBA_fnc_targetEvent;
    };
};
```